### PR TITLE
Update MongoDB to 4.2

### PR DESCRIPTION
**Pull request by klaussner**
To do:

- [x] Download MongoDB 4.2 in dev bundle scripts.
- [x] Use custom build for Linux (https://github.com/meteor/meteor-feature-requests/issues/361).
- [x] Test on macOS, Windows, and 32-bit Linux

### CODE REVIEW

File: packages/npm-mongo/package.js [hunk 1]
```diff
@@ -3,12 +3,12 @@
...
 Package.describe({
   summary: "Wrapper around the mongo npm package",
-  version: "3.1.2",
+  version: "3.3.0",
   documentation: null
 });
 
 Npm.depends({
-  mongodb: "3.1.13"
+  mongodb: "3.3.2"
```
Comment (benjamn) [inline packages/npm-mongo/package.js]: Can you update the shrinkwrap file for the `npm-mongo` package by running `meteor test-packages ./npm-mongo` in the `meteor/packages` directory?

**Comment by rj-david** [comment]
Just curious why meteor did not go the docker route for mongodb? More work for devs?

**Comment by klaussner** [comment]
I've updated all URLs to the latest versions. For now, the custom Linux build is downloaded from the [GitHub releases of my `mongodb-builder` repo](https://github.com/klaussner/mongodb-builder/releases).

**Comment by klaussner** [comment]
Regarding the custom MongoDB build: Should I transfer the `mongodb-builder` repo to the Meteor organization? Then the Meteor team could build the binaries and host them on GitHub when a new MongoDB version is released. Or do you prefer a different way to build and host the files? Just let me know and I'll update this PR with the new URLs. 🙂

**Comment by filipenevola** [comment]
Sure @klaussner , go ahead and transfer the repo to Meteor org, I'll provide write access to you also!

**Comment by filipenevola** [comment]
`mongodb-builder` repo is now under meteor org https://github.com/meteor/mongodb-builder

File: packages/npm-mongo/package.js [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
 Package.describe({
   summary: "Wrapper around the mongo npm package",
-  version: "3.3.0-beta190.4",
+  version: "3.4.0-beta190.4",
   documentation: null
 });
 
 Npm.depends({
-  mongodb: "3.2.7"
+  mongodb: "3.4.0"
```
Comment (benjamn) [inline packages/npm-mongo/package.js]: Note: I took the liberty of updating the `mongodb` npm package to the latest version (@klaussner had used 3.3.2 in the original version of this PR). Maybe that was not a good idea?

Comment (filipenevola) [reply packages/npm-mongo/package.js]: This is a good idea 😉 I believe @klaussner was going to do the same.

**Comment by klaussner** [comment]
I haven't figured out yet why the "mongo with multiple --port numbers" test doesn't pass. MongoDB outputs file locking errors when the port is changed. :thinking:

```
2019-12-21T14:04:36.151+0100 I  CONTROL  [initandlisten] MongoDB starting : pid=19516 port=4001 dbpath=<APP>/.meteor/local/db 64-bit host=arch
2019-12-21T14:04:36.151+0100 I  CONTROL  [initandlisten] db version v4.2.1
2019-12-21T14:04:36.151+0100 I  CONTROL  [initandlisten] git version: edf6d45851c0b9ee15548f0f847df141764a317e
2019-12-21T14:04:36.151+0100 I  CONTROL  [initandlisten] allocator: tcmalloc
2019-12-21T14:04:36.151+0100 I  CONTROL  [initandlisten] modules: none
2019-12-21T14:04:36.151+0100 I  CONTROL  [initandlisten] build environment:
2019-12-21T14:04:36.151+0100 I  CONTROL  [initandlisten]     distarch: x86_64
2019-12-21T14:04:36.151+0100 I  CONTROL  [initandlisten]     target_arch: x86_64
2019-12-21T14:04:36.151+0100 I  CONTROL  [initandlisten] options: { net: { bindIp: "127.0.0.1", port: 4001 }, replication: { oplogSizeMB: 8, replSet: "meteor" }, security: { authorization: "disabled" }, storage: { dbPath: "<APP>/.meteor/local/db" } }
2019-12-21T14:04:36.151+0100 I  STORAGE  [initandlisten] Detected data files in <APP>/.meteor/local/db created by the 'wiredTiger' storage engine, so setting the active storage engine to 'wiredTiger'.
2019-12-21T14:04:36.151+0100 I  STORAGE  [initandlisten] 
2019-12-21T14:04:36.151+0100 I  STORAGE  [initandlisten] ** WARNING: Using the XFS filesystem is strongly recommended with the WiredTiger storage engine
2019-12-21T14:04:36.151+0100 I  STORAGE  [initandlisten] **          See http://dochub.mongodb.org/core/prodnotes-filesystem
2019-12-21T14:04:36.151+0100 I  STORAGE  [initandlisten] wiredtiger_open config: create,cache_size=7471M,cache_overflow=(file_max=0M),session_max=33000,eviction=(threads_min=4,threads_max=4),config_base=false,statistics=(fast),log=(enabled=true,archive=true,path=journal,compressor=snappy),file_manager=(close_idle_time=100000,close_scan_interval=10,close_handle_minimum=250),statistics_log=(wait=0),verbose=[recovery_progress,checkpoint_progress],
2019-12-21T14:04:36.609+0100 E  STORAGE  [initandlisten] WiredTiger error (11) [1576933476:609245][19516:0x7f83d4b3a340], wiredtiger_open: __posix_file_lock, 391: <APP>/.meteor/local/db/WiredTiger.lock: handle-lock: fcntl: Resource temporarily unavailable Raw: [1576933476:609245][19516:0x7f83d4b3a340], wiredtiger_open: __posix_file_lock, 391: <APP>/.meteor/local/db/WiredTiger.lock: handle-lock: fcntl: Resource temporarily unavailable
2019-12-21T14:04:36.609+0100 E  STORAGE  [initandlisten] WiredTiger error (16) [1576933476:609286][19516:0x7f83d4b3a340], wiredtiger_open: __conn_single, 1682: WiredTiger database is already being managed by another process: Device or resource busy Raw: [1576933476:609286][19516:0x7f83d4b3a340], wiredtiger_open: __conn_single, 1682: WiredTiger database is already being managed by another process: Device or resource busy
2019-12-21T14:04:36.609+0100 E  STORAGE  [initandlisten] WiredTiger error (11) [1576933476:609381][19516:0x7f83d4b3a340], wiredtiger_open: __posix_file_lock, 391: <APP>/.meteor/local/db/WiredTiger.lock: handle-lock: fcntl: Resource temporarily unavailable Raw: [1576933476:609381][19516:0x7f83d4b3a340], wiredtiger_open: __posix_file_lock, 391: <APP>/.meteor/local/db/WiredTiger.lock: handle-lock: fcntl: Resource temporarily unavailable
2019-12-21T14:04:36.609+0100 E  STORAGE  [initandlisten] WiredTiger error (16) [1576933476:609391][19516:0x7f83d4b3a340], wiredtiger_open: __conn_single, 1682: WiredTiger database is already being managed by another process: Device or resource busy Raw: [1576933476:609391][19516:0x7f83d4b3a340], wiredtiger_open: __conn_single, 1682: WiredTiger database is already being managed by another process: Device or resource busy
2019-12-21T14:04:36.609+0100 E  STORAGE  [initandlisten] WiredTiger error (11) [1576933476:609517][19516:0x7f83d4b3a340], wiredtiger_open: __posix_file_lock, 391: <APP>/.meteor/local/db/WiredTiger.lock: handle-lock: fcntl: Resource temporarily unavailable Raw: [1576933476:609517][19516:0x7f83d4b3a340], wiredtiger_open: __posix_file_lock, 391: <APP>/.meteor/local/db/WiredTiger.lock: handle-lock: fcntl: Resource temporarily unavailable
2019-12-21T14:04:36.609+0100 E  STORAGE  [initandlisten] WiredTiger error (16) [1576933476:609525][19516:0x7f83d4b3a340], wiredtiger_open: __conn_single, 1682: WiredTiger database is already being managed by another process: Device or resource busy Raw: [1576933476:609525][19516:0x7f83d4b3a340], wiredtiger_open: __conn_single, 1682: WiredTiger database is already being managed by another process: Device or resource busy
```

**Comment by klaussner** [comment]
I found out what the problem was. The test runs Meteor multiple times in a row but `mongod` didn't shut down before the next process attempted to acquire the lock. Starting with version 4.0.8/4.1.10, MongoDB performs a step down procedure if the primary receives a SIGTERM signal (https://jira.mongodb.org/browse/SERVER-38994). During this procedure, the process doesn't shut down for up to ten seconds until a secondary becomes the new primary. Since Meteor starts a single-node replica set, this is unnecessary because there are no secondaries. Setting the `waitForStepDownOnNonCommandShutdown` parameter to `false` when starting MongoDB disables the step down. This will be the default for single-node replica sets in MongoDB 4.3 (relevant commit: https://git.io/JeNkT), so the parameter can be removed in the future. Hopefully, the test passes now. :crossed_fingers:

I also removed the `useNewUrlParser` option in 83e087b878283bb95599e9b200aa9f3e09ce0f11 because it [defaults to `true` since `mongodb` 3.3.3](https://github.com/mongodb/node-mongodb-native/blob/0a2d4e9e62c689942c91e16797f9d5570351c65d/HISTORY.md#333-2019-10-16).

File: packages/npm-mongo/package.js [hunk 3]
```diff
@@ -3,12 +3,12 @@
 
 Package.describe({
   summary: "Wrapper around the mongo npm package",
-  version: "3.3.0-rc190.0",
+  version: "3.4.0-rc190.1",
```
Comment (klaussner) [inline packages/npm-mongo/package.js]: Is this the right version number?

**Comment by klaussner** [comment]
The Travis CI build fails because the custom `mongod` binary links to a newer version of `libstdc++` than is installed in the Travis CI environment (which runs Ubuntu 16.04). I think this can be solved by running the custom build with an older Docker base image or by statically linking `libstdc++`. :thinking:

**Comment by klaussner** [comment]
I changed the build script to [link `libstdc++` statically](https://github.com/meteor/mongodb-builder/commit/d7e09ed320eb647750101db7e52988fb01b912ac) and created a [new release](https://github.com/meteor/mongodb-builder/releases/tag/4.2.1). The link to the archive is the same as before but we need a new dev bundle. @benjamn :eyes:

**Pull request merged**